### PR TITLE
feat(swap): multichain support

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

Here comes multichain swaps ⛓️ 

Using the new `/getSwapQuote` endpoint.

Couple of notes:
- fees are really high (we display the max fee) on Ethereum
  - squid wasn't giving a competitive quote (3-4x times higher in gas fee than others via MetaMask or Rainbow)
- uncovered a bug in the TX returned by squid (no `value` set when swapping a native asset)
- doesn't show a user friendly error when trying to swap assets accross chains (will be addressed separately)

### Test plan

- Updated tests
- Manual tests, didn't yet try to go through the actual swap on Ethereum given the current high fees.

**Ethereum (ETH to USDC and USDC to ETH)**

https://github.com/valora-inc/wallet/assets/57791/57f8dff6-aab7-4fa2-8590-ded06cf9e4c1

**Celo (CELO to agEUR and cUSD to axlUSDC)**

https://github.com/valora-inc/wallet/assets/57791/e75bcef7-b375-46ab-8377-f3209a88cde3

### Related issues

- Fixes RET-911

### Backwards compatibility

Yes.

### CODE REVIEW

File: src/swap/SwapScreen.tsx [hunk 1]
```diff
@@ -105,7 +105,7 @@ export function SwapScreen({ route }: Props) {
   const useViemForSwap = getFeatureGate(StatsigFeatureGates.USE_VIEM_FOR_SWAP)
 
   // sorted by USD balance and then alphabetical
-  const supportedTokens = useSelector(swappableTokensSelector)
```
Comment (kathaypacific) [inline src/swap/SwapScreen.tsx:108]: let's delete this selector? ✂️

**Review by kathaypacific — APPROVED** [review]
ah, so excited to see this!! i was wondering the same thing as Charlie, can we just axe the analytics properties that track token address? i'm not strongly opposed to keeping on sending this data but i can't think of how useful this data is when we have the tokenId now

File: src/analytics/Properties.tsx [hunk 2]
```diff
@@ -1220,8 +1220,22 @@ interface CoinbasePayEventsProperties {
 }
 
 interface SwapEvent {
-  toToken: string
-  fromToken: string
+  /**
+   * Address of the to token
+   *
+   * @deprecated use toTokenId instead
```
[the remark this answers is not in the document]
Comment (jeanregisser) [reply src/analytics/Properties.tsx:1226]: Not really, I just didn't want to break existing usage of this field. This comment is more for folks looking at analytics.

File: src/swap/SwapScreen.tsx [hunk 3]
```diff
@@ -230,13 +230,25 @@ export function SwapScreen({ route }: Props) {
...
+      }
+
       if (
         !exchangeRate.estimatedPriceImpact ||
         exchangeRate.estimatedPriceImpact.gte(priceImpactWarningThreshold)
       ) {
         ValoraAnalytics.track(SwapEvents.swap_price_impact_warning_displayed, {
-          toToken: exchangeRate.toTokenAddress,
-          fromToken: exchangeRate.fromTokenAddress,
+          toToken: toToken.address,
```
[the remark this answers is not in the document]
Comment (jeanregisser) [reply src/swap/SwapScreen.tsx:246]: The comment was really for people looking at analytics props. To warn them that `toToken` / `fromToken` will be removed at some point.

**Comment by jeanregisser** [comment]
> ah, so excited to see this!! i was wondering the same thing as Charlie, can we just axe the analytics properties that track token address? i'm not strongly opposed to keeping on sending this data but i can't think of how useful this data is when we have the tokenId now

We need to wait before removing it, this is still behind a feature flag and I don't want to break existing dashboard just now. Plus allows to see if we still incorrectly send `0xEeeeEEE...`

File: src/swap/SwapScreen.tsx (hunk 1 above)
Comment (jeanregisser) [reply src/swap/SwapScreen.tsx:108]: Good idea! 👍

**Pull request merged**